packages/core/admin/admin/src/components/LeftMenu.tsx

```diff
@@ -134,19 +134,29 @@ const LeftMenu = ({ generalSectionLinks, pluginsSectionLinks }: LeftMenuProps) =
             })}
           >
             {pluginsSectionLinks.map((link) => {
-              const Icon = link.icon;
+              const LinkIcon = link.icon;
+              const { lockIcon = false } = link;
 
               return (
-                <NavLink
-                  as={RouterNavLink}
-                  // @ts-expect-error the props from the passed as prop are not inferred // joined together
-                  to={link.to}
-                  key={link.to}
-                  icon={<Icon />}
-                  onClick={() => handleClickOnLink(link.to)}
-                >
-                  {formatMessage(link.intlLabel)}
-                </NavLink>
+                <Flex key={link.to} justifyContent="space-between">
+                  <NavLink
+                    as={RouterNavLink}
+                    // @ts-expect-error the props from the passed as prop are not inferred // joined together
+                    to={link.to}
+                    icon={<LinkIcon />}
+                    onClick={() => handleClickOnLink(link.to)}
+                  >
+                    {formatMessage(link.intlLabel)}
+                  </NavLink>
+                  {lockIcon && (
+                    <Icon
+                      width={`${15 / 16}rem`}
+                      height={`${15 / 16}rem`}
+                      as={Lock}
+                      marginRight={4}
+                    />
+                  )}
+                </Flex>
```

Comment:
i think the entire NavLink row still needs clicking incl. the lock icon, so instead of wrapping it in flex, the icon probably wants to sit inside the `NavLink`

Reply:
The NavLink accept only a string as a Child and not a combination of text and a component, it is because of that I preferred to add the icon as external

Reply:
i've just looked at the code for the component, did you try using the `badgeContent` prop? i'm not sure if it'll work. Otherwise i would still do it and ts ignore the type, it should be looser to be honest.

Reply:
https://github.com/strapi/design-system/blob/ee1a927b101b0a469f365da2fedcb5369acaae4e/packages/strapi-design-system/src/v2/MainNav/NavLink.tsx#L94
and
https://github.com/strapi/design-system/blob/ee1a927b101b0a469f365da2fedcb5369acaae4e/packages/strapi-design-system/src/v2/MainNav/NavLink.tsx#L105

Reply:
> did you try using the badgeContent prop

Reply:
Ok to fix the TS errors I used the badgeContent, but I had to add a div and wrap the NavLink to remove the badgeContent background (I tried styling directly the NavLink but then I had problems with the react-router-dom NavLink). In the future we can change a little bit the NavLink component in the DS to accept also Icon and not just text and also to add variants for the badge background color